### ISSUE TITLE
Post Author Name Block: theme.json settings don't work

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#39095]
### Description

When you try to change the appearance of the post-author-name block using theme.json, the changes don't work. This is because the block is loaded after the global styles compat files, so it doesn't know about the existence of the block yet. The solution might be to load all the blocks before the other compat files.

### Step-by-step reproduction instructions

Add this to your theme.json file for your theme:
```
			"core/post-author-name": {
				"typography": {
					"textTransform": "uppercase"
				}
			},
```

Add a post author name block to your post
Confirm that the block is not upper case

### Environment info

Gutenberg trunk

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
I am sorry I don't understand what this means :)
_the block is loaded after the global styles compat files, so it doesn't know about the existence of the block yet._
-Why would any block be loaded after that?

When adding the block, I placed it in the wrong "list" and I am trying to solve that here, is that related?
https://github.com/WordPress/gutenberg/pull/38320

**Comment by scruffian** [comment]
> I placed it in the wrong "list" and I am trying to solve that here, is that related?

Yes! This is fixed by https://github.com/WordPress/gutenberg/pull/38320

**Comment by scruffian** [comment]
I think there was something wrong with my testing, this is still an issue...

**Comment by carolinan** [comment]
@scruffian Who might we be able to ping to try get this looked at for 12.9 / 13 ?

**Comment by scruffian** [comment]
It's on my list of things to look at :)

**Comment by youknowriad** [comment]
This is fixed I think. Let me know if I missed anything in my testing.
